### PR TITLE
Hotfix/autoconf gnu extensions config

### PR DESCRIPTION
**Pull request by weatherman2095pro**
Ensure that GNU extensions are properly managed by `autoconf`, error-out if the extensions are unavailable.

`libstatsdc` requires the extensions for logging errors and doesn't have any conditional fallbacks in cases where they aren't found. The "error-out" behavior should be removed once proper fallbacks are implemented.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -16,7 +19,7 @@ AC_TYPE_SSIZE_T
 
 # Checks for library functions.
 AC_FUNC_MALLOC
-AC_CHECK_FUNCS([socket strdup strerror])
+AC_CHECK_FUNCS([socket strdup strerror vasprintf asprintf], [], [exit 1])
```
Comment (weatherman2095pro) [inline configure.ac:22]: I'm not fully sure `exit 1` is the right way to make it abort on error. I'm open to suggestions.

**Review by psyomn — APPROVED** [review]
Look OK. Can you double check that these changes don't affect `archgear` in any negative way (pls)?

**Review by pjhades — APPROVED** [review]
I never used autoconf before, but `exit 1` looks OK to me.

**Comment by weatherman2095pro** [comment]
> Look OK. Can you double check that these changes don't affect `archgear` in any negative way (pls)?

Aside from obvious need to change checksums, nothing broke.

**Comment by RAttab** [comment]
Could you explain what the problem you're solving is? Right now it's difficult to verify the fix without knowing what it is that we're fixing. It generally makes everyone's life easier to the PR  message contains a bit of context on the change.

**Comment by weatherman2095pro** [comment]
> Could you explain what the problem you're solving is? Right now it's difficult to verify the fix without knowing what it is that we're fixing. It generally makes everyone's life easier to the PR message contains a bit of context on the change.

Essentially, the prior config didn't use the macro required to enable GNU extensions, despite the program *requiring* them, resulting in compile-time warnings for me.

**Pull request merged**